src/test/java/karuna/karuna_backend/post/domain/PostServiceTest.java

```diff
@@ -26,4 +32,35 @@ void shouldCreatePost() {
         assertEquals(Constants.BODY, postDto.body());
     }
 
+    @Test
+    void shouldGetPosts() {
+        //given: Create the posts
+        PostCreateDto postCreateDto1 = new PostCreateDto("Body1");
+        PostCreateDto postCreateDto2 = new PostCreateDto("Body2");
+        PostCreateDto postCreateDto3 = new PostCreateDto("Body3");
+        PostDto postDto1 = postService.createPost(postCreateDto1);
+        PostDto postDto2 = postService.createPost(postCreateDto2);
+        PostDto postDto3 = postService.createPost(postCreateDto3);
+        //and: Set up pagination and sorting
+        Pageable pageable = PageRequest.of(0, 10, Sort.by("ID").descending());
+        //when: Fetch the posts
+        PostWrapper posts = postService.getPosts(pageable);
+        List<PostDto> listPost = posts.posts();
+        //then: Verify the results
+        assertNotNull(listPost);
+        assertEquals(3, listPost.size());
+        assertEquals("Body3", listPost.get(0).body());
+        assertEquals("Body2", listPost.get(1).body());
+        assertEquals("Body1", listPost.get(2).body());
+    }
+
+    @Test
+    void shouldNotGetPosts() {
+        //given: Set up pagination and sorting
+        Pageable pageable = PageRequest.of(0, 10, Sort.by("ID").descending());
+        //when: Get posts from non-existent cache
+        PostWrapper posts = postService.getPosts(pageable);
+        //then Check if not exist posts
+        assertTrue(posts.posts().isEmpty());
+    }
```

Comment:
@BarSwi what u think about this tests? what extacly is checked here @KrystynaCal ?

Reply:
Do we really need to check the scenario if getPosts would return some kind of "trash" if database is empty? I cannot imagine a scenario when that would be really neccessary but I might be wrong. Feel free to persuade me

Reply:
it's not my idea with this test. Deleting the cache didn't work in my test, so  I looked at another endpoint in the code how was this problem resolved.  And  I  recreated the logic

Reply:
![image](https://github.com/karunaproject/be/assets/110590007/9676328f-39fe-4d94-a13b-0397bf13953b)

Reply:
After merge my refactor u write tests in spock to this, for now look ok :D @KrystynaCal